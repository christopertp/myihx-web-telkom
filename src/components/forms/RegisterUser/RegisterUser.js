import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import PasswordField from '../../fields/Password';
import TextField from '../../fields/Text';
import styles from './styles.scoped.css';

export default function RegisterUser(props) {
  const { handleSubmit, initialize, invalid } = props;
  const { isLoading, userData } = useSelector(s => s.register);

  useEffect(() => {
    initialize(userData);
  }, []);

  const renderField = (name, label, placeholder, type) => {
    const Component = name === 'password' ? PasswordField : TextField;
    const inputProps = {
      label,
      placeholder,
      type,
    };

    return <Field component={Component} inputProps={inputProps} name={name} />;
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      {renderField('fullName', 'NAMA LENGKAP', 'Masukkan nama lengkap')}
      {renderField('mobileNumber', 'NOMOR PONSEL', 'E.g 081xxxxx or 62xxxxx', 'tel')}
      {renderField('email', 'EMAIL', 'Masukkan alamat email', 'email')}
      {renderField('password', 'KATA SANDI', 'Buat kata sandi baru', 'password')}
      <section className={styles['ref-box']}>
        {renderField('referralCode', 'Punya kode referral? Masukkan kode referral dan dapatkan hadiah spesial untuk Anda dan teman, serta agen kami!', 'Masukkan kode referral')}
      </section>
      <Button disabled={invalid || isLoading} type="submit">
        {isLoading ? <Spinner /> : 'Daftar Sekarang'}
      </Button>
    </form>
  );
}

RegisterUser.defaultProps = {
  handleSubmit: () => { },
  initialize: () => { },
  invalid: false,
};

RegisterUser.propTypes = {
  handleSubmit: PropTypes.func,
  initialize: PropTypes.func,
  invalid: PropTypes.bool,
};
