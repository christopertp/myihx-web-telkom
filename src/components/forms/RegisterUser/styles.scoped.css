.root > button {
  margin-top: 2rem;
  width: 100%;
}

.root > label:not(:first-of-type) {
  margin-top: 1.5rem;
}

.root > input {
  margin-top: 0.5rem;
}

.error {
  color: #ee4c24;
  display: block;
  font-size: 0.75rem;
  margin-top: 1rem;
}

.ref-box {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  margin-top: 1.5rem;
  padding: 1.5rem;
  word-wrap: break-word;
}

.ref-box > label {
  color: #333435;
  font-weight: bolder;
  font-size: 1rem;
  letter-spacing: normal;
  line-height: 1.5rem;
  text-transform: none !important;
}
