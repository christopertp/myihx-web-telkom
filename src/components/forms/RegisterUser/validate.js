import { isEmail, isPhone, isPassword } from '../../../utils/validation';

export default function validate(values) {
  const { email, fullName, mobileNumber, password } = values;

  return {
    email: !isEmail(email) ? 'Mohon cantumkan email yang aktif' : '',
    mobileNumber: !isPhone(mobileNumber) ? 'Mohon cantumkan nomor ponsel yang aktif' : '',
    fullName: !fullName ? 'Mohon cantumkan nama lengkap Anda' : '',
    password: !isPassword(password) ? 'Minimal 6 karakter terdiri dari gabungan huruf dan angka' : '',
  };
}
