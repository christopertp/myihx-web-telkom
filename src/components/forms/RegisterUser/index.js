import { reduxForm } from 'redux-form';
import RegisterUser from './RegisterUser' ;
import validate from './validate';

export default reduxForm({
  initialValues: {
    email: '',
    mobileNumber: '',
    fullName: '',
    password: '',
    referralCode: '',
  },
  form: 'registerUser',
  validate,
})(RegisterUser);
