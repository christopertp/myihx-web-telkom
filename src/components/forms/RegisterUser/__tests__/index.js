import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import RegisterUser from '../RegisterUser';

useEffect.mockImplementation(fn => fn());
useSelector.mockImplementation(fn => {
  fn(({ register: '' }));
  return { isLoading: false, message: '', userData: '' };
});

describe('src/components/forms/RegisterUser', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<RegisterUser />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: true, message: 'error', userData: { fullName: 'test-name' } }));
    RegisterUser.defaultProps.handleSubmit();
    const result = RegisterUser({
      ...RegisterUser.defaultProps,
    });
    const button = result.props.children[5];
    expect(button.props.children.type).toBe(Spinner);
  });
});
