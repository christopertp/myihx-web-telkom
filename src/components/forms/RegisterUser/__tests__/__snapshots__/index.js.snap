// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/RegisterUser render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "NAMA LENGKAP",
        "placeholder": "Masukkan nama lengkap",
        "type": undefined,
      }
    }
    name="fullName"
  />
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "NOMOR PONSEL",
        "placeholder": "E.g 081xxxxx or 62xxxxx",
        "type": "tel",
      }
    }
    name="mobileNumber"
  />
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "EMAIL",
        "placeholder": "Masukkan alamat email",
        "type": "email",
      }
    }
    name="email"
  />
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "KATA SANDI",
        "placeholder": "Buat kata sandi baru",
        "type": "password",
      }
    }
    name="password"
  />
  <section>
    <Field
      component={[Function]}
      inputProps={
        Object {
          "label": "Punya kode referral? Masukkan kode referral dan dapatkan hadiah spesial untuk Anda dan teman, serta agen kami!",
          "placeholder": "Masukkan kode referral",
          "type": undefined,
        }
      }
      name="referralCode"
    />
  </section>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Daftar Sekarang
  </Button>
</form>
`;
